macros/src/main/scala/org/zalando/grafter/macros/ReaderMacro.scala

```diff
@@ -0,0 +1,64 @@
+package org.zalando.grafter.macros
+
+import scala.annotation.StaticAnnotation
+import scala.language.experimental.macros
+
+object ReaderMacro {
+
+  def impl(c: scala.reflect.macros.whitebox.Context)(annottees: c.Expr[Any]*): c.Expr[Any] = {
+    import c.universe._
+
+    val inputs : (Tree, Tree, Option[Tree]) =
+      annottees.toList match {
+        case classDecl :: companion :: rest =>
+          (classDecl.tree, c.typecheck(classDecl.tree), Option(companion.tree))
+
+        case classDecl :: rest =>
```

Comment:
I think 

    case classDecl :: Nil 

would be clearer here, since there is no actual rest.